presto-main/src/main/java/com/facebook/presto/execution/controller/PidTaskExecutorController.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.execution.controller;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+public class PidTaskExecutorController
+        implements TaskExectorController
+{
+    private final PidController controller;
+    private final double target;
+
+    public PidTaskExecutorController(double target, int minValue, int maxValue, double kp, double ki, double kd)
+    {
+        checkArgument(minValue > 0, "minValue must be greater than 0");
+        checkArgument(minValue <= maxValue, "minValue must be less than maxValue");
+        checkArgument(target >= 0 && target <= 1, "target must be in range of [0, 1]");
+
+        this.target = target;
+        this.controller = new PidController(minValue, maxValue, target, kp, ki, kd);
+    }
+
+    @Override
+    public int getNextRunnerThreads(Optional<TaskExecutorStatistics> statistics)
+    {
+        requireNonNull(statistics, "statistics is null");
+
+        if (statistics.isPresent() && statistics.get().getWallTime().getValue(TimeUnit.SECONDS) >= 0.05) {
```

Comment:
The wall time being less than 0.05 sec should never happen because of your min property value, right?